ddl/ddl_worker.go

```diff
@@ -464,3 +476,64 @@ func updateSchemaVersion(t *meta.Meta, job *model.Job) (int64, error) {
 	err = t.SetSchemaDiff(diff)
 	return schemaVersion, errors.Trace(err)
 }
+
+func (d *ddl) cleanAddIndexQueueJobs(txn kv.Transaction) error {
+	startTime := time.Now()
+	m := meta.NewMeta(txn)
+	m.SetJobListKey(meta.AddIndexJobListKey)
+	for {
+		job, err := d.getFirstDDLJob(m)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if job == nil {
+			log.Infof("[ddl] cleaning jobs in the adding index queue takes time %v.", time.Since(startTime))
+			return nil
+		}
+		log.Infof("[ddl] cleaning job %v in the adding index queue.", job)
+
+		// The types of these jobs must be ActionAddIndex.
+		if job.SchemaState == model.StatePublic || job.SchemaState == model.StateNone {
```

Comment:
Do we have to rollback or cancel the add index ddl job? What about enqueueing these jobs to default job list

Reply:
I think canceling the job It's simpler and safer.

Reply:
If we just put all the jobs at the end of the default list, the worst-case result would be?

Reply:
@coocood 
Added an index that should not be added. 
e.g.
original order: drop column col1 -> add index idx1(col1) `should be failed` -> add column col1
current order: drop column col1 -> add column col1 -> add index idx1(col1) `successful `